src/pipelines/rust_app.rs

```diff
@@ -352,15 +352,28 @@ pub struct RustAppOutput {
 
 impl RustAppOutput {
     pub async fn finalize(self, dom: &mut Document) -> Result<()> {
+        let (base, js, wasm, head) = (&self.cfg.public_url, &self.js_output, &self.wasm_output, "html head");
+
+        let with_preload = format!(
+            r#"
+<link rel="preload" href="{base}{wasm}" as="fetch" type="application/wasm" crossorigin>
+<link rel="preload" href="{base}{js}" as="script" type="text/javascript" crossorigin>{head}"#,
+            base = base,
+            js = js,
+            wasm = wasm,
+            head = dom.select(head).html(),
+        );
+        dom.select(head).set_html(with_preload);
+
         let script = format!(
             r#"<script type="module">import init from '{base}{js}';init('{base}{wasm}');</script>"#,
-            base = self.cfg.public_url,
-            js = &self.js_output,
-            wasm = &self.wasm_output,
+            base = base,
+            js = js,
+            wasm = wasm,
         );
         match self.id {
             Some(id) => dom.select(&super::trunk_id_selector(id)).replace_with_html(script),
-            None => dom.select("html head").append_html(script),
+            None => dom.select(head).append_html(script),
```

Comment:
Based on [Martin's feedback here](https://github.com/thedodd/trunk/issues/145#issuecomment-796722855), I'm thinking that we should go ahead and actually append this to the end of body instead. That will also make it so that we don't have to include that update in the README about ensuring that the `rust-app` link is declared last.

Thoughts?

Reply:
I suppose we could add the script to the end of the body rather than the head, but this is in the case where the user hasn't provided a `rust-app` link. If he has, we should replace it with the script (which we do). The comment in the `README` is supposed to be a helpful note indicating that it's a good idea to put it after anything else that needs to load before the app.

Reply:
So, actually due to the way browsers load assets — namely the fact that they don't execute JS until assets have been loaded — the simple fact that we have a preload directive here for our JS/WASM means that we do not have to worry about the ordering of the assets in the HTML head at all.